src/main/java/com/ebivariation/contigalias/service/AssemblyService.java

```diff
@@ -52,54 +54,68 @@ public AssemblyService(AssemblyRepository repository, @Qualifier("NCBIDataSource
         this.dataSource = dataSource;
     }
 
-    public Optional<AssemblyEntity> getAssemblyOrFetchByAccession(String accession) throws IOException {
-        Optional<AssemblyEntity> assembly = getAssemblyByAccession(accession);
-        if (assembly.isPresent()) {
-            return assembly;
-        }
-        Optional<AssemblyEntity> fetchAssembly = fetchAndInsertAssembly(accession);
-        if (fetchAssembly.isPresent()) {
-            stripAssemblyFromChromosomes(fetchAssembly.get());
-            return fetchAssembly;
+    public List<AssemblyEntity> getAssemblyOrFetchByAccession(String accession) throws IOException {
```

Comment:
here again, I don't understand why we should return a list when querying by accession. This makes sense for searching by taxonomy.